libfulltext/doi/crossref/test_elsevier.py

```diff
@@ -0,0 +1,40 @@
+"""Unit tests for Elsevier"""
+
+from unittest import TestCase, skip
+
+from ...test.utils import assert_stream
+from .elsevier import get_elsevier_fulltext
+
+class GetElsevierFulltextTest(TestCase):
+    """Test get_elsevier_fulltext"""
+
+    @skip('Elsevier API currently does not return full texts but only page 1 '
+          '(re-enable with a working API key)')
+    @staticmethod
+    def test_sha1():
+        """Compare SHA1 and filename"""
+        get_elsevier_fulltext(
+            '10.1016/j.physletb.2016.07.042',
+            assert_stream('4724fea61643131e32dd4267608f977ffeafb70e', 'fulltext.pdf'),
+            apikey='f7840794d3322d4b56a9cf687aecfccb'
```

Comment:
Are you sure you want to insert your API key in the clear here?

I would rather only add it as an *encrypted* key to Travis and skip the test in case no key is found (e.g. a user running the test on his machine did not add the key to his config). I am happy to do the Travis fiddling as I did this a while back for something else.

Reply:
Yeah, I've also set this up a couple of times, no prob for me to do it. What do you think about reading env vars in our config, e.g. `PUBLISHERS_ELSEVIER_APIKEY`? I think this has 2 benefits:
* easy travis CI config
* this service can be used directly in a docker environment (where configuration is done via env vars)